zcash_client_sqlite/src/wallet.rs

```diff
@@ -357,20 +333,27 @@ pub(crate) fn ufvk_to_uivk<P: consensus::Parameters>(
 pub(crate) fn add_account<P: consensus::Parameters>(
     conn: &rusqlite::Transaction,
     params: &P,
-    account: Account,
+    kind: AccountType,
+    viewing_key: ViewingKey,
```

Comment:
Hm, doing it this way means that the `(Derived, IVK)` combination is supported, which doesn't make much sense.

Reply:
It's not an _invalid_ combination: one could import a UIVK for which you know its derivation pathway, in order to make future identification of the corresponding UFVK and USK easier.

Reply:
Yeah, I decided that that would be a reasonable use case, hence "this is fine."

Reply:
What this boils down to is that the thing that makes an account "usable as a wallet" is that it has a UFVK (and can track balance), not whether we know the derivation path; as long as the caller can provide the corresponding USK when needed, that's all that matters.